src/NuGet.Core/NuGet.Commands/PackCommandRunner.cs

```diff
@@ -217,40 +217,38 @@ private void AddOutputFiles(PackageBuilder builder, bool includeSymbols)
                 projectOutputDirectory = targetPath.Substring(0, targetPath.IndexOf(configFolderPath) + configFolderPath.Length);
             }
 
-            NuGetFramework targetFramework = NuGetFramework.AnyFramework;
-
-            // List of extensions to allow in the output path
-            var allowedOutputExtensions = new HashSet<string>(StringComparer.OrdinalIgnoreCase) {
-                ".dll",
-                ".exe",
-                ".xml",
-                ".winmd",
-                ".runtimeconfig.json",
-
-            };
-
-            if (includeSymbols)
+            if (!Directory.Exists(projectOutputDirectory))
             {
-                // Include pdbs for symbol packages
-                allowedOutputExtensions.Add(".pdb");
+                throw new Exception("No build found in " + projectOutputDirectory + ". Use the -Build option or build the project.");
             }
 
+            NuGetFramework targetFramework = NuGetFramework.AnyFramework;
             var targetFileName = Path.GetFileNameWithoutExtension(targetPath);
+            var outputFileNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+            {
+                $"{targetFileName}.dll",
+                $"{targetFileName}.exe",
+                $"{targetFileName}.xml",
+                $"{targetFileName}.winmd",
+                $"{targetFileName}.runtimeconfig.json"
```

Comment:
runtimeconfig.json will be along side .dll for each target framework which would need to copied as well.